utilities/tools/Set-AVMModule.ps1

```diff
@@ -85,28 +102,43 @@ function Set-AVMModule {
     # # Load helper scripts
     . (Join-Path $PSScriptRoot 'helper' 'Set-ModuleFileAndFolderSetup.ps1')
 
-    $resolvedPath = (Resolve-Path $ModuleFolderPath).Path
+    if ($InvokeForDiff) {
+        . (Join-Path $RepoRoot 'utilities' 'pipelines' 'sharedScripts' 'Get-GitDiff.ps1')
 
-    # Build up module file & folder structure if not yet existing. Should only run if an actual module path was provided (and not any of their parent paths)
-    if (-not $SkipFileAndFolderSetup -and (($resolvedPath -split '[\\|\/]avm[\\|\/](res|ptn|utl)[\\|\/].+?[\\|\/].+').count -gt 1)) {
-        if ($PSCmdlet.ShouldProcess("File & folder structure for path [$resolvedPath]", 'Setup')) {
-            Set-ModuleFileAndFolderSetup -FullModuleFolderPath $resolvedPath
+        $relevantTemplatePaths = Get-GitDiff -PathOnly | Where-Object { $_ -match '[\/|\\]main\.bicep$' }
```

Comment:
Should, if child-modules are identified (e.g., if only the secret module is updated), run for the parents as well.

Reply:
-> Filter for unique